mesa/time.py

```diff
@@ -142,14 +142,13 @@ class SimultaneousActivation(BaseScheduler):
 
     def step(self) -> None:
         """Step all agents, then advance them."""
-        agent_keys = list(self._agents.keys())
-        for agent_key in agent_keys:
-            self._agents[agent_key].step()
-        # We recompute the keys because some agents might have been removed in
-        # the previous loop.
-        agent_keys = list(self._agents.keys())
-        for agent_key in agent_keys:
-            self._agents[agent_key].advance()
+        agent_values = self._agents.values()
+        for agent in agent_values:
+            agent.step()
+        # the previous steps might remove some agents, but
+        # this loop will go over the remaining existing agents
+        for agent in agent_values:
```

Comment:
Why not just `for agent in self._agents.values()` for both? With the current version, it hasn't been proven that the generator is auto-updated when some agents are removed in `step()`.

Reply:
mmhh I'm not following you, I did it with `agent_values = self._agents.values()` so that not to call again the `dict.values()` again, it can have some side effects in your opinion? I think that it's okay because `agent_values` is a reference to a view object so it updates automatically

Reply:
`dict.values()` returns a generator-like object in Python 3, so it's a low-cost operation.

> I think that it's okay because agent_values is a reference to a view object so it updates automatically

Need a proof just in case it doesn't introduce regression.

Reply:
tested the cost of `dict.values()` -> really low cost, so I put the way you thought